### PR TITLE
Change `configure` to create prefix directories

### PR DESCRIPTION
**Pull request by brandon-neth**
Changes `./configure` to create the prefix directory if it doesn't exit. Also changes --print-chpl-home tests so they account for builds that use a prefix directory.

Completes #19762

### CODE REVIEW

File: configure [hunk 1]
```diff
@@ -89,6 +89,12 @@ mynormalizepath () {
       ;;
   esac
 
+  if [ ! -d $P ]
+  then
+    echo "Prefix directory does not exist, creating directory."
+    mkdir $P
```
Comment (mppf) [inline configure:95]: Maybe it should use `mkdir -p` to construct intervening paths as well?

**Review by mppf — APPROVED** [review]
Can you give the PR a better title? The title doesn't say it's updating `configure`.

I know you've run testing on this, but I like to put what was tested in the PR message, in order to communicate with my reviewers. And I like that too when I am reviewing. Not a hard requirement though.

**Comment by brandon-neth** [comment]
I tested it by configuring for a directory that didn't exist:
```
rm -r $HOME/NewDir
./configure --prefix=$HOME/NewDir
make -j8
make install
```

 Is there a way to add this test to the test suite? I'm not sure if build tests are in there since we build the compiler before running the tests.

File: configure (hunk 1 above)
Comment (brandon-neth) [reply configure]: done

**Review by mppf — APPROVED** [review]

**Comment by brandon-neth** [comment]
makes sense. Thanks!

**Pull request merged**